src/SignupModal.jsx

```diff
@@ -26,6 +26,34 @@ export const SIGNUP_MODAL_EMAIL_CLASS = `${SIGNUP_MODAL_CLASS}-email`;
  * @returns {React.element} SignupModal
  */
 export const SignupModal = ({ onDismiss, signupOptions, ...other }) => {
+	const facebookOnClick = () => {
```

Comment:
I'm not sure tracking is something that we'd want to address in each component. Is there more context to tracking GA in shared components or a way we've historically done this?

Reply:
Yeah, not sure if adding tracking like this to a shared component makes sense.

It's probably better to have additional click handler props in this component and manage tracking in the application where it's used.

Reply:
cc: @mmcgahan

Reply:
as a follow up i factored it out to be prop functions that consumers can pass in to do tracking in their own apps.

Reply:
Not sure we want to have specific props here for these functions. This specific type of configuration is hard to maintain and not an intuitive API for users. Changes like this should really wait to be merged in.